lineapy/api/models/pipeline.py

```diff
@@ -0,0 +1,187 @@
+        self.artifact_names: List[str] = artifacts
+        self.id = get_new_id()
+
+    def export(
+        self,
+        framework: str = "SCRIPT",
+        output_dir: Optional[str] = None,
+        pipeline_dag_config: Optional[AirflowDagConfig] = {},
+    ) -> Path:
+        execution_context = get_context()
+        db = execution_context.executor.db
+        session_orm = (
+            db.session.query(SessionContextORM)
+            .order_by(SessionContextORM.creation_time.desc())
+            .all()
+        )
+        if len(session_orm) == 0:
+            track(ExceptionEvent(ErrorType.PIPELINE, "No session found in DB"))
+            raise Exception("No sessions found in the database.")
+        last_session = session_orm[0]
+
+        if framework in PipelineType.__members__:
+            if PipelineType[framework] == PipelineType.AIRFLOW:
+
+                ret = AirflowPlugin(db, last_session.id).sliced_airflow_dag(
+                    self.name,
+                    self.task_graph,
+                    output_dir=output_dir,
+                    airflow_dag_config=pipeline_dag_config,
+                )
+
+            else:
+
+                ret = ScriptPlugin(db, last_session.id).sliced_pipeline_dag(
+                    self.name,
+                    self.task_graph,
+                    output_dir=output_dir,
+                )
+
+            # send the info
+            track(
+                ToPipelineEvent(
+                    framework,
+                    len(self.artifact_names),
+                    self.task_graph.get_airflow_dependency() != "",
+                    pipeline_dag_config is not None,
+                )
+            )
+            return ret
+
+        else:
+            raise Exception(f"No PipelineType for {framework}")
+
+    def export2(
+        self,
+        framework: str = "SCRIPT",
+        output_dir: str = ".",
+        pipeline_dag_config: Optional[AirflowDagConfig2] = {},
+    ) -> None:
+        """
+        This method uses objects implementing new style of graph refactor.
+        It is meant to eventually replace existing `export()` method above.
+        TODO: Replace existing `export()` method with this one.
+        """
+        # Create artifact collection
+        execution_context = get_context()
+        artifact_collection = ArtifactCollection(
+            execution_context.executor.db, self.artifact_names
+        )
+
+        # Check if the specified framework is a supported/valid one
+        if framework not in PipelineType.__members__:
+            raise Exception(f"No PipelineType for {framework}")
+
+        # Determine writer class to use
+        pipeline_writer_class = {
+            PipelineType.SCRIPT: BasePipelineWriter,
+            PipelineType.AIRFLOW: AirflowPipelineWriter,
+        }[PipelineType[framework]]
+
+        # Construct pipeline writer
+        pipeline_writer = pipeline_writer_class(
```

Comment:
This is probably a different PR, but I want to tackle the naming and structure here too. 
 
`pipeline_writer_class` is effectively a factory. If you think that making a factory class is too overkill at this point I would at least suggest renaming this to `pipeline_writer_factory`
```

Reply:
Done